### ISSUE TITLE
tests: new DB migration hooks called with NULL pointer

### ISSUE THREAD

**Issue by m-schmoock** [ElementsProject/lightning#2442]
### Issue and Steps to Reproduce
Using the new (smarter) DB migration update routine currently requires a early return in all hooks as certain unit tests call the hooks with a `*ld` NULL pointer.

### Workaround
```C
void migrate_hook(struct lightningd *ld, struct db *db)
{
       /* Note: Some unit tests call migrate without *ld context:
        * - wallet/test/run-wallet
        * - wallet/test/run-db
        * We need to return so we dont get nullpointer access violation. */
       if (!ld) return;

       // your update here
```

### Solution
I don't know how to solve this in a clean way

**Comment by m-schmoock** [comment]
@rustyrussell 

For the `wallet/test/run-wallet.c` its easy to add the pointer, but not for the `wallet/test/run-db.c` and probably future test cases. We could change the `static void db_migrate(struct lightningd *ld, struct db *db, struct log *log)` in `wallet/db.c` to only call the migration hooks if *ln was set.

Not elegant, but maybe sufficient:

### Workaround:
Change this:
```C
		if (dbmigrations[current].func)
			dbmigrations[current].func(ld, db);
```

To this:
```C

		if (dbmigrations[current].func && ld)
			dbmigrations[current].func(ld, db);
```

### Benefits:
No need to early return on every migration hook.

### Drawbacks: 
A test that is not supplying *ld cannot rely on the migrations to be applied correctly.

**Comment by cdecker** [comment]
Closed by #2444
